webui/src/default/page-settings.tsx

```diff
@@ -89,6 +91,95 @@ export default function createPageSettings(theme: Theme, prefersDarkMode: boolea
     const additionalRoutes: React.FunctionComponent = () =>
         <Route path='/about' render={() => <About />} />;
 
+    const headTags: React.FunctionComponent<{title?: string, description?: string, keywords?: string, url?: string, imageUrl?: string, type?: string}> = (props) => {
+        const handleChangeClientState = (newState: any, addedTags: HelmetTags, removedTags: HelmetTags): void => {
+            if (addedTags.metaTags) {
+                addedTags.metaTags.forEach((value: HTMLMetaElement) => {
+                    if (!value.content) {
+                        value.remove();
+                    }
+                });
+            }
+        };
+
+        const twitterCard = props.imageUrl ? 'summary_large_image' : 'summary';
+        const type = props.type || 'website';
+        return <Helmet onChangeClientState={handleChangeClientState}>
+            <title>{props.title}</title>
+
+            {/* SEO Meta Tags */}
+            <meta name='description' content={props.description}/>
```

Comment:
This will render meta tags with an empty `content` attribute depending on the props passed. It might be better to just not render these at all when a prop is `undefined`. However, I believe it doesn't affect SEO in a negative way so it may not be a big deal.

Reply:
@oliviergoulet5 I tried that at first, but when a meta tag isn't rendered then `react-helmet` keeps the meta tag from the parent page.
**Home page:**
```
<meta itemProp='description' content='Open VSX is an Eclipse open-source project.'/>
<meta itemProp='image' content='https://open-vsx.org/openvsx-preview.png'/>
```
**Extension page:**
The extension has a `description` but no `image`, the result is:
```
<meta itemProp='description' content='YAML Language Support by Red Hat'/>
<meta itemProp='image' content='https://open-vsx.org/openvsx-preview.png'/>
```
When `undefined` is passed then you get:
```
<meta itemProp='description' content='YAML Language Support by Red Hat'/>
<meta itemProp='image' content=''/>
```
The `onChangeClientState` handler then filters out meta tags with empty `content` attribute:
https://github.com/eclipse/openvsx/blob/7f6f7fcab5033d0a218596fed78c5e4ae8673e66/webui/src/default/page-settings.tsx#L95-L103